ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -111,6 +111,7 @@ run tracers chainDbTracer diffusionTracers diffusionArguments networkMagic dbPat
       (nodeProtocolMagicId (Proxy @blk) cfg)
     withRegistry $ \registry -> do
 
+      lockDbMarkerFile registry dbPath
```

Comment:
I first thought that the functionality should be integrated into `checkDbMarker`, but that function is parametric in the file-system, and `HasFS` doesn't support file-locking.

The cleanest solution would be to add file-locking support to `HasFS`, but I don't think it is worth the effort, as we're not really going to use it in the tests.

So I think it is fine as is.